format.py

```diff
@@ -29,7 +29,7 @@
 
 # Compiler options used with Clang-Tidy
 # Normal warnings are disabled with -Wno-everything to focus only on tidying
-INCLUDES = "-Iinclude -Isrc -Ibuild -I."
+INCLUDES = "-Iinclude -Isrc -Ibuild/gc-eu-mq-dbg -I."
```

Comment:
Unfortunately always including build/ of a single version means you expect that version to be built which may not be the case (e.g. you may be building a different version than this one without having ever built the version used for format)

Note I am not sure if it is even relevant to include build/ for formatting, afaik the only reason for including build/ in general is for assets including .inc.c files, and the formatter isn't run on assets, but I may be forgetting stuff

Reply:
Hmm it seems to work find even if this directory does not exist. Should I remove this include? (What about in the `.vscode` docs?)

Reply:
For vscode settings it's useful because it makes the ide find those .inc.c files

I'm not sure what to do about formatting (I think the script silences all errors btw?)
I'll look into it and report back

Reply:
Iirc clang-tidy needs proper includes otherwise it will error instead of doing what it needs to with the file

Reply:
What about adding a check in format.py to fail if build/gc-eu-mq-dbg does not exist? I kinda think gc-eu-mq-dbg should be the "default" version for formatting, since it should be standardized but maybe clang-tidy might make different decisions per-version (though I'm not sure why it would)

Reply:
this is only an issue for the purpose of using the repo tools for different versions, which we postponed to another PR, so we can sweep this under the rug for now and be content with hardcoding gc-eu-mq-dbg for formatting

Reply:
Alright, I made a note to myself about this